crates/ika-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -81,10 +81,70 @@ pub enum PublicInput {
 #[derive(Clone)]
 pub struct MPCEventData {
     pub private_input: MPCPrivateInput,
-    pub init_protocol_data: MPCProtocolInitData,
-    pub(crate) decryption_shares: HashMap<PartyID, <AsyncProtocol as Protocol>::DecryptionKeyShare>,
+    pub request_input: MPCRequestInput,
+    pub(crate) decryption_key_shares:
```

Comment:
If something is optional, it should be `Option`, not an empty `HashMap`, this is dangerous.